aea/test_tools/test_skill.py

```diff
@@ -0,0 +1,368 @@
+
+        incoming_message = message_type(**message_attributes)
+        incoming_message.sender = sender
+        incoming_message.to = (
+            self.skill.skill_context.agent_address if to is None else to
+        )
+
+        return incoming_message
+
+    def build_incoming_message_for_dialogue(
+        self,
+        dialogue: Dialogue,
+        performative: Message.Performative,
+        message_type: Optional[Type[Message]] = None,
+        dialogue_reference: Optional[Tuple[str, str]] = None,
+        message_id: Optional[int] = None,
+        target: Optional[int] = None,
+        to: Optional[Address] = None,
+        sender: Address = None,
+        **kwargs,
+    ) -> Message:
+        """
+        Quickly create an incoming message with the provided attributes for a dialogue.
+
+        For any attribute not provided, a value based on the dialogue is used.
+        These values are shown in parantheses in the list of parameters below.
+
+        NOTE: This method must be used with care. The dialogue provided is part of the skill
+        which is being tested. Because for any unspecified attribute, a "correct" value is used,
+        the test will be, by design, insured to pass on these values.
+
+        :param dialogue: the dialogue to which the incoming message is intended
+        :param performative: the performative of the message
+        :param message_type: (the message_class of the provided dialogue) the type of the message
+        :param dialogue_reference: (the dialogue_reference of the provided dialogue) the dialogue reference of the message
+        :param message_id: (the id of the last message in the provided dialogue + 1) the id of the message
+        :param target: (the id of the last message in the provided dialogue) the target of the message
+        :param to: (the agent address associated with this skill) the receiver of the message
+        :param sender: (the counterperty in the provided dialogue) the sender of the message
+        :param kwargs: other attributes
+
+        :return: the created incoming message
+        """
+        if dialogue is None:
+            raise AEAEnforceError("dialogue cannot be None.")
+
+        if dialogue.last_message is None:
+            raise AEAEnforceError("dialogue cannot be empty.")
+
+        message_type = (
+            message_type
+            if message_type is not None
+            else dialogue._message_class  # pylint: disable=protected-access
+        )
+        dialogue_reference = (
+            dialogue_reference
+            if dialogue_reference is not None
+            else dialogue.dialogue_label.dialogue_reference
+        )
+        message_id = (
+            message_id
+            if message_id is not None
+            else dialogue.last_message.message_id + 1
+        )
+        target = target if target is not None else dialogue.last_message.message_id
+        to = to if to is not None else self.skill.skill_context.agent_address
```

Comment:
although correct, we should be consistent and use the dialogue here:
```suggestion
        to = to if to is not None else dialogue.self_address
```

Reply:
Done